src/datastore/postgres/schema/v6.ts

```diff
@@ -0,0 +1,34 @@
+import { IDatabase } from "pg-promise";
+import { MatrixUser } from "matrix-appservice-bridge";
+
+// tslint:disable-next-line: no-any
+export async function runSchema(db: IDatabase<any>) {
+    // Create database
+    await db.none(`
+        CREATE TABLE linked_accounts (
+            user_id TEXT NOT NULL,
+            slack_id TEXT NOT NULL,
+            team_id TEXT NOT NULL,
+            team_name TEXT NOT NULL,
+            access_token TEXT NOT NULL,
+            CONSTRAINT cons_linked_accounts_unique UNIQUE(user_id, slack_id)
+        );
+    `);
+    // Insert entries from users table.
+    const users = await db.manyOrNone("SELECT userid, json FROM users WHERE isremote = false;");
+    for (const userData of users) {
+        const user = new MatrixUser(userData.userid, JSON.parse(userData.json));
+        if (!user.get("accounts")) {
+            continue;
+        }
+        // tslint:disable-next-line: no-any
+        for (const [slack_id, account] of Object.entries<any>(user.get("accounts") || { })) {
+            await db.none("INSERT INTO linked_accounts VALUES (${user_id}, ${slack_id}, ${team_id}, ${team_name})", {
+                user_id: userData.userid,
+                slack_id,
+                team_id: account.team_id,
+                team_name: account.team_name,
+            });
+        }
```

Comment:
You could do all this in only [one insert](https://stackoverflow.com/a/37302557) :

```suggestion
    const pgp = require('pg-promise')()  # tihs line is probably wrong
    cons cs = new pgp.helpers.ColumnSet(['user_id', 'slack_id', 'team_id', 'team_name'], {table: 'linked_accounts'};
    let values = [];

    for (const userData of users) {
        const user = new MatrixUser(userData.userid, JSON.parse(userData.json));
        if (!user.get("accounts")) {
            continue;
        }
        // tslint:disable-next-line: no-any
        for (const [slack_id, account] of Object.entries<any>(user.get("accounts") || { })) {
            values.push(
                {
                    user_id: userData.userid,
                    slack_id,
                    team_id: account.team_id,
                    team_name: account.team_name,
                }
            );
        }
    }
    
    const query = pgp.helpers.insert(values, cs);
    await db.none(query);
```

Disclaimer: I am definitively not a typescript dev :p

Reply:
Nice find, I am doing this one :)